### PR TITLE
Thymeleaf forms

### PR DESCRIPTION
**Pull request by jamietiblis**
I started setting up the edit and delete features but am having an issue on line 120 in StudentController. I wanted to see if you could take a look at it and I could try to get the edit and delete features set up this evening.

### CODE REVIEW

File: src/main/java/com/example/demo/StudentController.java [hunk 1]
```diff
@@ -1,52 +1,146 @@
...
+
+        // save updated student object
+        return "redirect:/students";
+    }
+
+
+
+
+    // CODE BELOW IS FOR DELETING A STUDENT
+
```
Comment (aldencaron) [inline src/main/java/com/example/demo/StudentController.java:140]: Add @Transactional annotation to this method, so 
```
@Transactional
@GetMapping("/students/delete/{studentId}")
public String deleteStudent(@PathVariable Integer studentId) {
        studentRepository.deleteStudentByStudentID(studentId);
        return "redirect:/students";
}
```

File: src/main/java/com/example/demo/StudentController.java [hunk 2]
```diff
@@ -1,52 +1,157 @@
...
+        }
+        if(existingStudent != null) {
+            model.addAttribute("student", (existingStudent));
+        }
+        return "editstudent";
+    }
+
+
+    @PostMapping("/students/{id}")
+    public String updateStudent(@PathVariable Integer studentId,
```
Comment (aldencaron) [inline src/main/java/com/example/demo/StudentController.java:126]: What you have in the {} needs to match what the variable name is.
So 
```
@PostMapping("/students/{studentId}")
public String updateStudent(@PathVariable Integer studentId, ...
```

File: src/main/java/com/example/demo/StudentController.java [hunk 3]
```diff
@@ -1,52 +1,157 @@
...
+        Optional<Student> studentOptional = studentRepository.findById(studentId);
+        Student existingStudent = null;
+        if(studentOptional.isPresent()){
+            existingStudent = studentOptional.get();
+        }
+        if(existingStudent != null){
+            existingStudent.setStudentID(studentId);
+            existingStudent.setFirstName(student.getFirstName());
+            existingStudent.setLastName(student.getLastName());
+            existingStudent = studentRepository.save(existingStudent);
```
Comment (aldencaron) [inline src/main/java/com/example/demo/StudentController.java:140]: You'll want to add sets here for student year and major as well if you want those editable.

File: src/main/resources/templates/editstudent.html [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                        <input
+                                type = "text"
+                                name = "lastName"
+                                th:field = "*{lastName}"
+                                class = "form-control"
+                                placeholder="Enter Student Last Name"
+                        />
+                    </div>
+
+                    <!--
```
[the remark this answers is not in the document]
Comment (jamietiblis) [reply src/main/resources/templates/editstudent.html]: Thank you for looking over the code, I fixed the typo on line 125 of StudentController, updated lines 140-141 to add sets for studentYear and major in StudentController, and uncommented lines 79-90 in editstudent. I'm able to see the delete feature, but still getting the 500 error when I try to interact with it. Are you able to interact with the delete feature on your end or are you getting the 500 error? I get the same error with the edit feature also.

[the remark this answers is not in the document]
Comment (aldencaron) [reply src/main/resources/templates/editstudent.html]: Yeah I was able to get it working. I can try again tomorrow with what you have now and see if it works for me.

[the remark this answers is not in the document]
Comment (jamietiblis) [reply src/main/resources/templates/editstudent.html]: Ok I added all the code for the edit and delete features in ProfessorController, ProfessorRepository, and the html templates editprofessor and professorsview, but when I run it I am getting a 400 bad request error in the browser. I don't have any errors in the terminal just in the browser. Do you have any suggestions on what could be causing this?

[the remark this answers is not in the document]
Comment (aldencaron) [reply src/main/resources/templates/editstudent.html]: Hello! First, the professor issue is a typo in the url. 
Then the foreign key constraint error you're getting sounds to me like the student id does not exist when you're trying to create an enrollment with that id. With foreign key constraints you need to be careful about how the databases relate to each other.

**Pull request merged**